lib/src/executables/intl_message_migration.dart

```diff
@@ -187,61 +159,123 @@ void main(List<String> args) async {
-
-    exitCode = await runCodemodSequences(packageDartPath, [
-      [intlPropMigrator],
-      [displayNameMigrator],
-      [importMigrator]
-    ]);
-
-    processedPackages.add(package);
-    if (exitCode != 0 || outputFile.readAsStringSync() == classPredicate) {
-      outputFile.deleteSync();
-    } else {
-      List<String> lines = outputFile.readAsLinesSync();
-      final functions = lines.sublist(4);
-      functions.removeWhere((string) => string == '');
-      functions.sort();
-      lines.replaceRange(4, lines.length, functions);
-      lines.add('}');
-      outputFile.writeAsStringSync(lines.join('\n'), mode: FileMode.write);
-    }
+    await migratePackage(
+        package, packageNameLookup, processedPackages, codemodArgs, dartPaths);
+  }
+}
+
+void printUsage() {
+  stderr.writeln(
+      'Migrates literal strings that seem user-visible in the package by wrapping them in Intl.message calls.');
+  stderr.writeln();
+  stderr.writeln('Usage:');
+  stderr.writeln('    intl_message_migration [arguments]');
+  stderr.writeln();
+  stderr.writeln('Options:');
+  stderr.writeln(parser.usage);
+}
+
+/// Runs a set of codemod sequences separately to work around an issue where
+/// updates from an earlier suggestor aren't reflected in the resolved AST
+/// for later suggestors.
+///
+/// This means we have to set up analysis contexts multiple times, which takes longer,
+/// but isn't a dealbreaker. E.g., for wdesk_sdk, running two sequences takes 2:52
+/// as opposed to 2:00 for one sequence.
+///
+/// If any sequence fails, returns that exit code and short-circuits the other
+/// sequences.
+Future<int> runCodemodSequences(
+  Iterable<String> paths,
+  Iterable<Iterable<Suggestor>> sequences,
+  List<String> codemodArgs,
+) async {
+  for (final sequence in sequences) {
+    final exitCode = await runInteractiveCodemodSequence(
+      paths,
+      sequence,
+      defaultYes: true,
+      args: codemodArgs,
+      additionalHelpOutput: parser.usage,
+    );
+    if (exitCode != 0) return exitCode;
+  }
+
+  return 0;
+}
+
+/// Migrate files included in [paths] within [package].
+///
+/// We expect [paths] to be absolute.
+Future<void> migratePackage(
+    String package,
+    Map<String, String> packageNameLookup,
+    Set<String> processedPackages,
+    List<String> codemodArgs,
+    List<String> paths) async {
+  _log.info('Starting migration for $package');
+
+  final packageRoot = p.basename(package);
+  final packageName = packageNameLookup[packageRoot] ?? 'fix_me_bad_name';
+  _log.info('Starting migration for $packageName');
+  List<String> packageDartPaths;
+  try {
+    packageDartPaths =
+        dartFilesToMigrateForPackage(package, processedPackages).toList();
+  } on FileSystemException {
+    _log.info('${package} does not have a lib directory, moving on...');
+    return;
+  }
+
+  // Limit the paths we handle to those that were included in [paths]
```

Comment:
If there were paths on the command line, limit ourselves to those. In theory, we could be doing multiple sub-packages and have some files from each on the command line, which is why we filter the package paths based on the arguments, instead of just using the arguments directly.